docs/proposals/decentralized-identity-management.md

```diff
@@ -0,0 +1,177 @@
+* Use SPIFFE specification based SPIRE for implementation of identity servers.
+
+### SPIFFE and SPIRE
+The Secure Production Identity Framework For Everyone (SPIFFE) Project defines a framework and set of standards for identifying and securing communications between services. SPIFFE enables enterprises to transform their security posture from just protecting the edge to consistently securing all inter-service communications deep within their applications. SPIFFE recommends industry standards like TLS and JWT for forming a identity document for every service. The service-level identity AuthN and AuthZ removes the dependency of complex network-level ACL strategies.
+
+More information about SPIFFE can be found at https://github.com/spiffe/spiffe.
+
+SPIRE (SPIFFE Runtime Environment) is a reference implementation of SPIFFE specification. SPIRE manages identities for node and workloads. It provides API for controlling attestation policies and identity issuance and rotation strategies.
+
+More information about SPIRE can be found at https://github.com/spiffe/spire.
+
+### Benefits
+Node attestation: Only verifiable edge nodes can join the edge clusters. Every node is issued an identity on verification. In case of failed node attestations, no identity documents can be issued for services running on the node.
+
+Workload attestation: Only verifiable workload can run on edge nodes. In case of failed workload attestations, there are no identities issues for the workloads. All communications are blocked from unverified workloads.
+
+Certificate rotation: Short-lived certificates are generated and rotation policies can be configured for every service communication. There is no need for custom agents and reliance on specific orchestrators for certificate rotation configuration and management.
+
+Automated non-root CA certificate heirarchical deployments: Edge spire servers can be configured to not share any root CA chain for downstream nodes and workloads.
+
+### Why not Kubernetes secrets?
+
+* Kubernetes secrets work in a centralized model of secret distribution. To support offline scenarios, usage of long-lived certificates are used. Usage of long-lived certificates expose keys and certificates for longer duration which will enable theft and therefore are a higher value to an attacker since a stolen key enables the attacker to impersonate the service.
+* Known risks of using Kubernetes secrets (here)[https://kubernetes.io/docs/concepts/configuration/secret/#risks]. The presented solution mitigates risks related to protection of secret by application, secret visibility to other instance of application, impersonation of application instance (though, this requires a stronger method of workload selector used for attestation) and accidental storage of secretes on shared volumes by storing in-memory. In-memory storage have issues but pose a harder problem. A better notification framework will be proposed in the forthcoming versions.
+
+## Architecture goal.
```

Comment:
For the architecture, I'd suggest add descriptions for each major components and their functionalities. Also why to have SPIRE server on edge and cloud. Some descriptions about the data flow would be helpful. 

In below section, you did explain about node/app. identity acquisition & attestation. Here I am suggesting that we add some high level overview. Specially how this design be able to achieve the benefits; and address the issue secrets hit.